### DISCUSSION TITLE
Use cache when refetching if a condition is met?

### DISCUSSION THREAD

**Discussion by 1Computer1** [TanStack/query#3522] [Q&A]
Hello, suppose I have two fetchers, `fetchLastModified` and `fetchExpensive`.
I would like to be able to do something like this:
```js
const { data: lastModified, isSuccess } = useQuery('lastModified', fetchLastModified);
const { data } = useQuery('expensive', fetchExpensive, {
  enabled: isSuccess,
  useCache: lastModified > lastLastModified, // Something like this?
});
```

The idea is that if nothing is in cache, it would fetch, but if there is something in cache and the condition is met, then it would not refetch (incl. on refocus and such). Perhaps there is a manual way to implement this logic?

Thanks!

**Comment by Moshyfawn** [comment]
Take a look at [this part of RQ docs](https://react-query.tanstack.com/guides/initial-query-data#using-initialdata-to-prepopulate-a-query=)

**Reply by 1Computer1** [reply]
That seems to only work on the initial fetch, not on subsequent refetches?

**Reply by TkDodo** [reply]
what do you mean subsequent refetches? If you pair `initialData` with `staleTime`, this should be pretty much what you need.
